<!--
 Software License Agreement (BSD License)
 Copyright Â© 2022-2023 belongs to Shadow Robot Company Ltd.
 All rights reserved.
 Redistribution and use in source and binary forms, with or without modification,
 are permitted provided that the following conditions are met:
   1. Redistributions of source code must retain the above copyright notice,
      this list of conditions and the following disclaimer.
   2. Redistributions in binary form must reproduce the above copyright notice,
      this list of conditions and the following disclaimer in the documentation
      and/or other materials provided with the distribution.
   3. Neither the name of Shadow Robot Company Ltd nor the names of its contributors
      may be used to endorse or promote products derived from this software without
      specific prior written permission.
 This software is provided by Shadow Robot Company Ltd "as is" and any express
 or implied warranties, including, but not limited to, the implied warranties of
 merchantability and fitness for a particular purpose are disclaimed. In no event
 shall the copyright holder be liable for any direct, indirect, incidental, special,
 exemplary, or consequential damages (including, but not limited to, procurement of
 substitute goods or services; loss of use, data, or profits; or business interruption)
 however caused and on any theory of liability, whether in contract, strict liability,
 or tort (including negligence or otherwise) arising in any way out of the use of this
 software, even if advised of the possibility of such damage.
-->

<launch>
    <!-- the semantic description that corresponds to the URDF-->
    <param name="robot_description_semantic" textfile="$(find sr_moveit_hand_config)/config/shadowhand_motor.srdf"/>
    <test test-name="hand_kinematics_plugin_test_ff" pkg="hand_kinematics" type="test_kinematics_as_plugin"
          name="right_hand_kinematics_ff" time-limit="180">
        <param name="tip_name" value="rh_fftip"/>
        <param name="root_name" value="rh_palm"/>
        <param name="first_finger/maxIterations" type="int" value="1000"/>
        <param name="first_finger/epsilon" type="double" value="0.001"/>
        <param name="search_discretization" value="0.025"/>
        <param name="number_fk_tests" value="100"/>
        <param name="number_ik_tests" value="100"/>
        <param name="number_fk_tests_with_callbacks" value="100"/>
        <param name="group_name" value="first_finger"/>
        <param name="plugin_name" value="hand_kinematics/HandKinematicsPlugin"/>
    </test>

    <test test-name="hand_kinematics_plugin_test_mf" pkg="hand_kinematics" type="test_kinematics_as_plugin"
          name="right_hand_kinematics_mf" time-limit="180">
        <param name="tip_name" value="rh_mftip"/>
        <param name="root_name" value="rh_palm"/>
        <param name="middle_finger/maxIterations" type="int" value="1000"/>
        <param name="middle_finger/epsilon" type="double" value="0.001"/>
        <param name="search_discretization" value="0.025"/>
        <param name="number_fk_tests" value="100"/>
        <param name="number_ik_tests" value="100"/>
        <param name="number_fk_tests_with_callbacks" value="100"/>
        <param name="group_name" value="middle_finger"/>
        <param name="plugin_name" value="hand_kinematics/HandKinematicsPlugin"/>
    </test>

    <test test-name="hand_kinematics_plugin_test_rf" pkg="hand_kinematics" type="test_kinematics_as_plugin"
          name="right_hand_kinematics_rf" time-limit="180">
        <param name="tip_name" value="rh_rftip"/>
        <param name="root_name" value="rh_palm"/>
        <param name="ring_finger/maxIterations" type="int" value="1000"/>
        <param name="ring_finger/epsilon" type="double" value="0.001"/>
        <param name="search_discretization" value="0.025"/>
        <param name="number_fk_tests" value="100"/>
        <param name="number_ik_tests" value="100"/>
        <param name="number_fk_tests_with_callbacks" value="100"/>
        <param name="group_name" value="ring_finger"/>
        <param name="plugin_name" value="hand_kinematics/HandKinematicsPlugin"/>
    </test>

    <test test-name="hand_kinematics_plugin_test_lf" pkg="hand_kinematics" type="test_kinematics_as_plugin"
          name="right_hand_kinematics_lf" time-limit="180">
        <param name="tip_name" value="rh_lftip"/>
        <param name="root_name" value="rh_palm"/>
        <param name="little_finger/maxIterations" type="int" value="1000"/>
        <param name="little_finger/epsilon" type="double" value="0.001"/>
        <param name="search_discretization" value="0.025"/>
        <param name="number_fk_tests" value="100"/>
        <param name="number_ik_tests" value="100"/>
        <param name="number_fk_tests_with_callbacks" value="100"/>
        <param name="group_name" value="little_finger"/>
        <param name="plugin_name" value="hand_kinematics/HandKinematicsPlugin"/>

    </test>
    <test test-name="hand_kinematics_plugin_test_th" pkg="hand_kinematics" type="test_kinematics_as_plugin"
          name="right_hand_kinematics_th" time-limit="180">
        <param name="tip_name" value="rh_thtip"/>
        <param name="root_name" value="rh_palm"/>
        <param name="thumb/maxIterations" type="int" value="1000"/>
        <param name="thumb/epsilon" type="double" value="0.001"/>
        <param name="search_discretization" value="0.025"/>
        <param name="number_fk_tests" value="100"/>
        <param name="number_ik_tests" value="100"/>
        <param name="number_fk_tests_with_callbacks" value="100"/>
        <param name="group_name" value="thumb"/>
        <param name="plugin_name" value="hand_kinematics/HandKinematicsPlugin"/>
    </test>
</launch>
